File: frontend/src/types/media.ts
/**
 * Something the creator generated. Mirrors backend/app/schemas/media.py.
 *
 * A finished video and a scene clip are kept as distinct kinds rather than
 * flattened: one is the deliverable, the other is a part that made it, and
 * the library is more useful when it can say which.
 */

export type MediaKind = "video" | "clip";

export type MediaItem = {
  id: string;
  kind: MediaKind;
  project_id: string;
  project_title: string;
  /** What this is within its project: "Finished video", "Scene 3 · take 2". */
  label: string;
  duration_seconds: number | null;
  file_size_bytes: number | null;
  /** Auth-gated proxy route, fetched through the API client. */
  url: string;
  created_at: string;
};

File: backend/app/schemas/media.py
import enum
import uuid
from datetime import datetime

from pydantic import BaseModel


class MediaKind(str, enum.Enum):
    """A finished video, or one scene clip that went into one."""

    video = "video"
    clip = "clip"


class MediaItemOut(BaseModel):
    id: uuid.UUID
    kind: MediaKind
    project_id: uuid.UUID
    project_title: str
    # What this is within its project: "Finished video", "Scene 3 · take 2".
    label: str
    duration_seconds: float | None = None
    file_size_bytes: int | None = None
    # Auth-gated proxy route; the client fetches it through the API client
    # rather than putting it straight in a src attribute.
    url: str
    created_at: datetime
